packages/webapp/src/containers/Finances/ExpenseDetail/index.jsx

```diff
@@ -1,222 +1,103 @@
-          <h4>{date}</h4>
-          <DropdownButton options={options}>
-            {this.props.t('SALE.EXPENSE_DETAIL.ACTION')}
-          </DropdownButton>
-        </div>
-
-        <div className={styles.itemContainer}>
-          <Semibold>{this.props.t('SALE.EXPENSE_DETAIL.DESCRIPTION')}</Semibold>
-          <div>{this.props.t('SALE.EXPENSE_DETAIL.COST')}</div>
-        </div>
-        {/* {expenseItems.length > 0 &&
-          expenseItems.map((e) => {
-            return (
-              <div key={e.type_name}>
-                <div className={styles.typeNameContainer}>
-                  <Semibold>{e.type_name}</Semibold>
-                </div>
-                {e.items.length > 0 &&
-                  e.items.map((i) => {
-                    return (
-                      <div key={i.note + i.value.toString()} className={styles.itemContainer}>
-                        <div>{'- ' + i.note}</div>
-                        <div className={styles.greenText}>
-                          {this.state.currencySymbol + i.value.toFixed(2).toString()}
-                        </div>
-                      </div>
-                    );
-                  })}
-              </div>
-            );
-          })} */}
-        <div key={expense.type}>
-          <div className={styles.typeNameContainer}>
-            <Semibold>{expense.type}</Semibold>
-          </div>
-          <div key={expense.note + expense.amount.toString()} className={styles.itemContainer}>
-            <div style={{ overflowWrap: 'anywhere' }}>{'- ' + expense.note}</div>
-            <div className={styles.greenText}>{expense.amount}</div>
-          </div>
-        </div>
-        {/* <div className={styles.itemContainer}>
-          <Semibold>{this.props.t('SALE.EXPENSE_DETAIL.TOTAL')}</Semibold>
-          <div className={styles.greenText} id="total-amount">
-            {this.state.currencySymbol + total}
-          </div>
-        </div> */}
-        <ConfirmModal
-          open={this.state.showModal}
-          onClose={() => this.setState({ showModal: false })}
-          onConfirm={() => {
-            this.deleteExpense();
-            this.setState({ showModal: false });
-          }}
-          message={this.props.t('SALE.EXPENSE_DETAIL.TEMP_DELETE_CONFIRMATION')}
-        />
-      </div>
-    );
-  }
-}
-
-const mapStateToProps = (state) => {
-  return {
-    // expense_detail_date: expenseDetailDateSelector(state),
-    expenses: expenseSelector(state),
-    expenseTypes: allExpenseTypeSelector(state),
-    farm: userFarmSelector(state),
-    expense: expenseToDetailSelector(state),
+
+  const onRetire = () => {
+    dispatch(deleteExpense(expense_id));
   };
-};
 
-const mapDispatchToProps = (dispatch) => {
-  return {
-    dispatch,
+  const handleGoBack = () => {
+    history.back();
   };
+
+  return (
+    expense && (
+      <PureExpenseDetail
+        key={isEditing ? 'editing' : 'readonly'} // remount the component
```

Comment:
Which trait is used for remounting? Does it behave differently compared to passing `view={isEditing ? 'edit' : 'read-only'}`?

Reply:
It's the `key` prop specifically! So if you pass just view and remove this line, the uncontrolled form inputs (having not unmounted) will store their value when you hit the back button, even if you don't hit submit! It's for that behaviour specifically I added the key.

Reply:
Oh I see! It's tricky to reuse the container 🤣 Wise solution!